### PR TITLE
docs(vrl): add documentation for `sieve` function

### PR DESCRIPTION
**Pull request by esensar**
Related: https://github.com/vectordotdev/vrl/pull/724

### CODE REVIEW

File: website/cue/reference/remap/functions/sieve.cue [hunk 1]
```diff
@@ -0,0 +1,72 @@
+package metadata
+
+remap: functions: sieve: {
+	category: "String"
+	description: """
+		Keeps only mathces of `pattern` in `value`.
```
Comment (jhgilbert) [inline website/cue/reference/remap/functions/sieve.cue:6]:
```suggestion
		Keeps only matches of `pattern` in `value`.
```

**Review by jhgilbert — APPROVED** [review]
Approved with one suggestion to fix a typo. Thank you!

File: website/cue/reference/remap/functions/sieve.cue (hunk 1 above)
Comment (esensar) [reply website/cue/reference/remap/functions/sieve.cue]: Thanks!
Although, I wonder how the spellchecker didn't catch this.

**Review by pront — APPROVED** [review]
To be merged after the next VRL update.

**Pull request merged**